src/lib/UI/Value/ValueFactory.php

```diff
@@ -158,16 +159,30 @@ public function createLanguage(Language $language, VersionInfo $versionInfo): UI
      *
      * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
      * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     * @throws \eZ\Publish\API\Repository\Exceptions\BadStateException
      */
     public function createRelation(Relation $relation, Content $content): UIValue\Content\Relation
     {
         $contentType = $content->getContentType();
         $fieldDefinition = $contentType->getFieldDefinition($relation->sourceFieldDefinitionIdentifier);
 
+        $contentInfo = $content->contentInfo;
+
+        try {
+            $relationLocation = $this->locationService->loadLocation($contentInfo->mainLocationId);
+        } catch (UnauthorizedException $e) {
+            // try different locations if main location is not accessible for the user
+            $relationLocations = $this->locationService->loadLocations($contentInfo);
+            if (empty($relationLocations)) {
+                throw $e;
```

Comment:
And what if the given content does not have a location?

Reply:
@andrerom Content without location is not a part of this JIRA ticket or PR. It ought to fix users being forbidden to load Content if they do not have access to main location but have access to other locations, not dealing with content without locations.